tests/unit/parsers/test.manifestjson.js

```diff
@@ -813,9 +814,46 @@ describe('ManifestJSONParser', () => {
 
         expect(manifestJSONParser.isValid).toEqual(true);
         const { warnings } = addonLinter.collector;
-        expect(warnings[0].code).toEqual(messages.MANIFEST_CSP.code);
+        expect(warnings[0].code).toEqual(messages.MANIFEST_CSP);
         expect(warnings[0].message).toContain('content_security_policy');
       });
+
+      // Manifest v3 formats.
+      invalidValues.forEach((invalidValue) => {
+        const addonLinter = new Linter({ _: ['bar'] });
+
+        const contentSecurityPolicy = {
+          extension_pages: invalidValue,
+          content_scripts: invalidValue,
+          // Alias for content_scripts.
+          isolated_world: invalidValue,
+        };
+
+        const jsonV3 = validManifestJSON({
+          content_security_policy: contentSecurityPolicy,
+          applications: {
+            // The new content_security_policy syntax is only supported
+            // on Firefox >= 72.
+            gecko: { strict_min_version: '72.0' },
+          },
+        });
+
+        const manifestV3JSONParser = new ManifestJSONParser(
+          jsonV3,
+          addonLinter.collector
+        );
+
+        expect(manifestV3JSONParser.isValid).toEqual(true);
+        const { warnings } = addonLinter.collector;
+
+        const keys = Object.keys(contentSecurityPolicy);
+        for (let i = 0; i < keys.length; i++) {
+          expect(warnings[i].code).toEqual(messages.MANIFEST_CSP);
+          expect(warnings[i].message).toContain(
+            `content_security_policy.${keys[i]}`
+          );
+        }
```

Comment:
Unless I'm mistaken we do expect exactly two warnings here, once for `content_scripts` and once for `isolated_worlds` - could we add an assertion for that too?

Reply:
No, the linter adds exactly 3 warnings, one for each of those properties ("extension_pages", "content_scripts" and "isolated_worlds").

It is true that `isolated_worlds` is used as an alias for `content_scripts` in Firefox, but that happens in the implementation side, for an API schema perspective they are 2 separate properties of the content_script_policy object.

Nevertheless I can add an assertion for the total number of warning we expect (which is 3 at the moment).